tmt/steps/execute/__init__.py

```diff
@@ -410,9 +410,12 @@ def prepare_tests(self, guest: Guest, logger: tmt.log.Logger) -> list[TestInvoca
             invocation = TestInvocation(phase=self, test=test, guest=guest, logger=logger)
             invocations.append(invocation)
 
+            # Exported metadata is the test's metadata along with other variables like the context
+            test_metadata = test._metadata
```

Comment:
`.copy()` is needed, otherwise, the line below would be the same as `test._metadata['context'] = ...`

Reply:
I thought that the moment the variable is altered it becomes a local variable, but maybe `__setitem__` does not count for that. Anyway better to be safe

Reply:
Nope. Python variables aren’t variables, just labels. There is a dictionary somewhere in memory, pointed to by `test._metadata`, after the assignment it’s still the same dict, but now `test_metadata` points to it too. No implicit `copy()`.